presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,548 @@ Tuning Presto
+
+.. _node-scheduler-network-topology:
+
+``node-scheduler.network-topology``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (``legacy`` or ``flat``)
+ * **Default value:** ``legacy``
+ * **Description:** Sets the network topology to use when scheduling splits. ``legacy`` will ignore the topology when scheduling splits. ``flat`` will try to schedule splits on the same host as the data is located by reserving 50% of the work queue for local splits. It is recommended to use ``flat`` for clusters where distributed storage runs on same nodes as presto workers.
+
+
+.. _tuning-pref-optimizer:
+
+Optimizer properties
+--------------------
+
+``optimizer.processing-optimization``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (``disabled``, ``columnar`` or ``columnar_dictionary``)
+ * **Default value:** ``disabled``
+ * **Description:** Serves as default value for ``processing_optimization`` session property. Setting this property will allow to set up optimization for filtering operators. Setting ``columnar`` allows to use columnar processing instead of using row by row to process the filter. Setting ``columnar_dictionary`` adds additional dictionary to simplify columnar scan. Setting this to value other than ``disabled`` is advised for data containing large rows often filtered by simple key.
+
+``optimizer.dictionary-aggregation``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** Serves as default value for ``dictionary_aggregation`` session property. Enables optimization for aggregations on dictionaries.
+
+
+``optimizer.optimize-hash-generation``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``true``
+ * **Description:** Serves as default value for ``optimize_hash_generation`` session property. Compute hash codes for distribution, joins, and aggregations early in query plan which may allow to drop some of computation later in query processing with the cost of increased preprocessing. In most cases it should decrease overall query processing time.
+
+
+``optimizer.optimize-metadata-queries``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** Enables optimization of aggregations that are kept in metadata of data structures. This allow to perform simple queries in ``O(1)`` time using metadata that are kept anyway. Currently this optimization can be use for selecting `max`, `min` and `approx_distinct` of partition keys. Using this may speed some queries significantly with possible drawback on very small data sets.
```

Comment:
Setting this property to ``true`` enables optimization of some aggregations by using values that are kept in metadata. This allows Presto to execute some simple queries in ``O(1)`` time. Currently this optimization applies to `max`, `min` and `approx_distinct` of partition keys. Using this may speed some queries significantly, though it may have a negative effect when used with very small data sets.